src/utils.hh

```diff
@@ -25,7 +25,7 @@ struct Matcher {
   Matcher(const std::string &pattern); // throw
   Matcher(Matcher &&) = default;
   ~Matcher();
-  bool matches(const std::string &text) const;
+  [[nodiscard]] bool matches(const std::string &text) const;
```

Comment:
How do you find places where `[[nodiscard]]` should be added? Did you just arbitrarily find some places and add `[[nodiscard]]`? (I am also less clear the attributes provides much value..)

Reply:
:rofl: They are diagnosed via `clang-tidy`